addons/forward/models.py

```diff
@@ -44,7 +44,7 @@ def save(self, request=None, *args, **kwargs):
             else:
                 user = request.user
 
-            self.owner.check_spam(user, {'addons_forward_node_settings__url'}, get_headers_from_request(request))
+            self.owner.check_spam(user, {'addons_forward_node_settings__url'}, get_headers_from_request(request), force=True)
```

Comment:
As per the last review, how does `force` address the following?

> cases where the `url` is added when the node is private, but isn't checked when it becomes public.

Reply:
Presumably that situation never produces spam because this happen on every model save regardless of whether it's public or private, but I see now this is broken by `if settings.SPAM_CHECK_PUBLIC_ONLY and not self.is_public:` but I should be able to override that one line and be good. I can also add something on ln 46 that checks if `owner.is_public`, hopefully that's on the right track.

Reply:
Bypassing the `SPAM_CHECK_PUBLIC_ONLY` check seems like the wrong path.

Why not just conditionally include `'addons_forward_node_settings__url'` in `get_spam_fields` `if node.has_addon('forward')`?

Reply:
> Why not just conditionally include 'addons_forward_node_settings__url' in get_spam_fields if node.has_addon('forward') ?

Yeah this makes sense, probably better. 

Just side comment though, as for `SPAM_CHECK_PUBLIC_ONLY` I don't see why this behavior is configurable here at all, our code seems to assume this is True implictly, like in `get_spam_fields`,

Reply:
>`if self.is_public and 'is_public' in saved_fields`

This causes it to check all potentially-spammy fields when something goes `private->public`, thereafter checking only changed fields.